index.js

```diff
@@ -275,20 +291,35 @@ function isHaltingViolation(violation, engine) {
  */
 async function writeComments() {
   console.log("Writing comments using GitHub REST API...");
-  const octokit = new Octokit();
-  const owner = this.pullRequest?.base?.repo?.owner?.login;
-  const repo = this.pullRequest?.base?.repo?.name;
+  const {octokit, owner, prNumber, repo} = getApi();
   for (let file in this.filePathToComments) {
     for (let comment of this.filePathToComments[file]) {
-      const method = `POST /repos/${owner}/${repo}/pulls/${this.pullRequest?.number}/comments`;
-      await octokit.request(method, comment);
+      // TODO: Add in resolving comments when the issue has been resolved?
+      const existingComment = this.existingComments.find(existingComment => matchComment(comment, existingComment));
+      if (!existingComment) {
+        const method = `POST /repos/${owner}/${repo}/pulls/${prNumber}/comments`;
+        await octokit.request(method, comment);
+      } else {
+        console.log('Skipping existing comment...', {
+          line: existingComment.line,
+          body: existingComment.body,
+          path: existingComment.path
```

Comment:
Printing the entire body to the log might get a little noisy. Is it possible to print the URL for the existing comment?

Reply:
Changed to `console.log(`Skipping existing comment ${existingComment.url}`);`